admin/src/Environment.js

```diff
@@ -49,29 +55,39 @@ class Environment extends Component {
 
     render() {
         const {
-            state: { config, edition, error, loading, newConfig },
+            state: { edition, error, loading, newConfig, filteredConfig },
             effects: { setNewConfig, toggleEdition },
+            setSearch,
         } = this.props;
 
         return (
             <div style={styles.container}>
                 <Card>
                     <CardText>
-                        {!edition && <RaisedButton label="Edit" primary onClick={toggleEdition} />}
-                        {edition && <RaisedButton label="Save" primary onClick={this.handleSaveClick} />}
-                        {edition && <RaisedButton label="Cancel" onClick={toggleEdition} />}
+                        {edition &&
+                            <div>
+                                <RaisedButton label="Save" primary onClick={this.handleSaveClick} />
+                                <RaisedButton label="Cancel" onClick={toggleEdition} />
+                            </div>}
+                        {!edition &&
+                            <div style={styles.actions}>
+                                <TextField hintText="search" onChange={setSearch} />
+                                <div style={styles.search}>
```

Comment:
If we do not wrap the RaisedButton in a div, its display get bugged by the display flex on the container.